src/engine/ResultTable.h

```diff
@@ -21,7 +21,7 @@ class ResultTable {
  public:
   enum Status { FINISHED = 0, OTHER = 1 };
 
-  enum class ResultType { KB, VERBATIM, TEXT };
+  enum class ResultType { KB, VERBATIM, TEXT, FLOAT, STRING };
```

Comment:
Can you add comments detailing what each type is (e.g. difference between `TEXT`, `STRING`, `KB` and `VERBATIM`)